Integration tests flaky

Sometimes integration tests fails with this error. We should investigate what could be happening.

```bash
KUBECONFIG='./.kubeconfig-dev' minikube update-context
* "minikube" context has been updated to point to 10.1.0.5[9](https://github.com/newrelic-forks/newrelic-prometheus/runs/7691440436?check_suite_focus=true#step:7:10):8443
* Current context is "minikube"
go test ./... -tags=integration_test -count=1 -race
?   	github.com/newrelic-forks/newrelic-prometheus/cmd/configurator	[no test files]
ok  	github.com/newrelic-forks/newrelic-prometheus/configurator	0.196s
ok  	github.com/newrelic-forks/newrelic-prometheus/configurator/kubernetes	0.024s
?   	github.com/newrelic-forks/newrelic-prometheus/configurator/promcfg	[no test files]
ok  	github.com/newrelic-forks/newrelic-prometheus/configurator/remotewrite	0.025s
ok  	github.com/newrelic-forks/newrelic-prometheus/configurator/statictargets	0.024s
2022/08/05 12:38:18 http: TLS handshake error from 127.0.0.1:60664: tls: first record does not look like a TLS handshake
2022/08/05 12:38:18 http: TLS handshake error from 127.0.0.1:60668: tls: first record does not look like a TLS handshake
2022/08/05 12:38:19 http: panic serving 127.0.0.1:60670: Fail in goroutine after Test_ExternalLabelsAreAddedToEachSample has completed
goroutine 114 [running]:
net/http.(*conn).serve.func1()
	/opt/hostedtoolcache/go/1.18.4/x64/src/net/http/server.go:1825 +0x[10](https://github.com/newrelic-forks/newrelic-prometheus/runs/7691440436?check_suite_focus=true#step:7:11)6
panic({0x2050020, 0xc000124240})
	/opt/hostedtoolcache/go/1.18.4/x64/src/runtime/panic.go:844 +0x258
testing.(*common).Fail(0xc000105ba0)
	/opt/hostedtoolcache/go/1.18.4/x64/src/testing/testing.go:817 +0x19b
testing.(*common).Errorf(0xc000105ba0, {0x227e552, 0x3}, {0xc000124200, 0x1, 0x1})
	/opt/hostedtoolcache/go/1.18.4/x64/src/testing/testing.go:934 +0xa7
github.com/stretchr/testify/assert.Fail({0x310d700, 0xc000105ba0}, {0xc00002c0a0, 0x97}, {0x0, 0x0, 0x0})
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.8.0/assert/assertions.go:264 +0x429
github.com/stretchr/testify/assert.NoError({0x310d700, 0xc000105ba0}, {0x310a160, 0xc000754460}, {0x0, 0x0, 0x0})
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.8.0/assert/assertions.go:1392 +0x125
github.com/stretchr/testify/require.NoError({0x3[11](https://github.com/newrelic-forks/newrelic-prometheus/runs/7691440436?check_suite_focus=true#step:7:12)3408, 0xc000105ba0}, {0x310a160, 0xc000754460}, {0x0, 0x0, 0x0})
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.8.0/require/require.go:[12](https://github.com/newrelic-forks/newrelic-prometheus/runs/7691440436?check_suite_focus=true#step:7:13)61 +0xa5
github.com/newrelic-forks/newrelic-prometheus/test/integration/mocks.handlerWithProxy.func1({0x311fb90, 0xc000538ee0}, 0xc0006c9d00)
	/home/runner/work/newrelic-prometheus/newrelic-prometheus/test/integration/mocks/remote_write.go:74 +0x6ca
net/http.HandlerFunc.ServeHTTP(0xc00072d320, {0x311fb90, 0xc000538ee0}, 0x1?)
	/opt/hostedtoolcache/go/1.18.4/x64/src/net/http/server.go:2084 +0x4e
net/http.serverHandler.ServeHTTP({0xc00080a9f0?}, {0x311fb90, 0xc000538ee0}, 0xc0006c9d00)
	/opt/hostedtoolcache/go/1.18.4/x64/src/net/http/server.go:29[16](https://github.com/newrelic-forks/newrelic-prometheus/runs/7691440436?check_suite_focus=true#step:7:17) +0x897
net/http.(*conn).serve(0xc0002a6500, {0x31205c8, 0xc000122f60})
	/opt/hostedtoolcache/go/1.[18](https://github.com/newrelic-forks/newrelic-prometheus/runs/7691440436?check_suite_focus=true#step:7:19).4/x64/src/net/http/server.go:[19](https://github.com/newrelic-forks/newrelic-prometheus/runs/7691440436?check_suite_focus=true#step:7:20)66 +0xbab
created by net/http.(*Server).Serve
	/opt/hostedtoolcache/go/1.18.4/x64/src/net/http/server.go:[30](https://github.com/newrelic-forks/newrelic-prometheus/runs/7691440436?check_suite_focus=true#step:7:31)71 +0x80d
FAIL
FAIL	github.com/newrelic-forks/newrelic-prometheus/test/integration	[35](https://github.com/newrelic-forks/newrelic-prometheus/runs/7691440436?check_suite_focus=true#step:7:36).049s
?   	github.com/newrelic-forks/newrelic-prometheus/test/integration/mocks	[no test files]
FAIL
make: *** [Makefile:70: integration-test] Error 1
```

--- a/test/integration/integration_kubernetes_filter_test.go
+++ b/test/integration/integration_kubernetes_filter_test.go
@@ -198,9 +198,10 @@ func addPodToEnv(t *testing.T, podMetadata *metadata, prefix string, k8sEnv k8sE
 		return nil
 	}
 
-	return k8sEnv.addPod(
+	return k8sEnv.addPodAndWaitOnPhase(
 		t,
 		fakePod(prefix, podMetadata.annotations, podMetadata.labels),
+		corev1.PodRunning,
 	)
 }
 
@@ -212,9 +213,10 @@ func addEndpointsToEnv(t *testing.T, endpointsMetadata *metadata, prefix string,
 	}
 
 	selector := map[string]string{"k8s.io/app": "myApp"}
-	k8sEnv.addPod(
+	k8sEnv.addPodAndWaitOnPhase(
 		t,
 		fakePod(prefix, map[string]string{}, selector),
+		corev1.PodRunning,
 	)
 
 	return k8sEnv.addService(

